Zend/zend_object_handlers.c

```diff
@@ -1020,9 +1020,13 @@ ZEND_API zval *zend_std_get_property_ptr_ptr(zend_object *zobj, zend_string *nam
 						ZVAL_NULL(retval);
 						zend_error(E_WARNING, "Undefined property: %s::$%s", ZSTR_VAL(zobj->ce->name), ZSTR_VAL(name));
 					}
-				} else if (prop_info && UNEXPECTED(prop_info->flags & ZEND_ACC_READONLY)
-					&& !verify_readonly_initialization_access(prop_info, zobj->ce, name, "initialize")) {
-					retval = &EG(error_zval);
+				} else if (prop_info && UNEXPECTED(prop_info->flags & ZEND_ACC_READONLY)) {
+					if (!verify_readonly_initialization_access(prop_info, zobj->ce, name, "initialize")) {
```

Comment:
Just wondering, is it possible to drop this check as well? Might as well always fall back.

Reply:
From what I understand, falling back to the `read`/`write` handlers will always trigger the `read` handler first, at which point it's impossible to know if the original operation was a write or read action. The result is still an exception, but with a different message (access to uninitialized property, rather than the scope access error). But maybe I'm completely missing what you're suggesting.

Reply:
Only write_property will be called for a pure write. And I think that something like `$this->x += 1` *should* be throwing an error about an uninitialized read rather incorrect initialization visibility, because this is what `$this->x = $this->x + 1` would do.

Reply:
That makes sense. So I'll move the `verify_readonly_initialization_access()` to `write_property`, but for `$this->foo[] = 'bar';` I'll still be lacking information in `read_property`, so we can either stick with the "uninitialized property" message or keep it in `get_property_ptr_ptr`, which one do you prefer?

Reply:
I think that one should result in an RW fetch, so if we wanted to customize the message, we probably could.